tasks/cephfs/filesystem.py

```diff
@@ -140,16 +140,22 @@ def clear_ceph_conf(self, subsys, key):
 
     def are_daemons_healthy(self):
         """
-        Return true if all daemons are in one of active, standby, standby-replay
+        Return true if all daemons are in one of active, standby, standby-replay, and
+        at least max_mds daemons are in 'active'.
+
         :return:
         """
+
+        active_count = 0
         status = self.mon_manager.get_mds_status_all()
         for mds_id, mds_status in status['info'].items():
             if mds_status['state'] not in ["up:active", "up:standby", "up:standby-replay"]:
                 log.warning("Unhealthy mds state {0}:{1}".format(mds_id, mds_status['state']))
                 return False
+            elif mds_status['state'] == 'up:active':
+                active_count += 1
 
-        return True
+        return active_count >= status['max_mds']
```

Comment:
Okay, so we can't actually have more MDSes active than max_mds. I guess if the states are changing while iterating we could count one active logical rank across two daemons, but in that case is the cluster really healthy?